File: frontend/src/main/typescript/backend/dto/request/songsByVocaDbEventTagRequest.ts
import { DbSortOrder } from "@/backend/dto/enumeration";
import { EventDateBounds } from "@/backend/dto/lowerLevelStruct";

export interface SongsByVocaDbEventTagRequest {
  tagId: number;
  startOffset: number;
  maxResults: number;
  orderBy: DbSortOrder;
  dates: EventDateBounds;
  clientType: string;
}

File: backend/src/main/kotlin/niconicotagger/dto/api/request/SongsByVocaDbEventTagRequest.kt
package niconicotagger.dto.api.request

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import jakarta.validation.constraints.Min
import niconicotagger.dto.api.misc.ClientType
import niconicotagger.dto.api.misc.EventDateBounds
import niconicotagger.dto.api.misc.VocaDbSortOrder
import org.hibernate.validator.constraints.Range

@JsonIgnoreProperties(ignoreUnknown = false)
data class SongsByVocaDbEventTagRequest(
    val tagId: Long,
    @field:Min(0) val startOffset: Long,
    @field:Range(min = 10, max = 100) val maxResults: Long,
    val orderBy: VocaDbSortOrder,
    val dates: EventDateBounds,
    val clientType: ClientType,
)
